linked_list.py

```diff
@@ -56,15 +60,12 @@ def search(self, val):
     def remove(self, node):
         # remove node from list, wherever it might be
         if self.size() == 0:
-            return "THE LIST! IT'S EMPTY!!"
+            raise ValueError("The list is empty")
         else:
             prevNode = None
             currentNode = self.firstNode
-            foundNode = False
-            while not foundNode:
-                if currentNode == node:
-                    foundNode = True
-                elif currentNode is None:
+            while currentNode is not node:
+                if currentNode is None:
```

Comment:
Though this is much much cleaner, it's still a bit over-done.  As you move forward through the other linked-list-related data structures, see if you can't find more efficient, more readable ways of accomplishing this task.